### PR TITLE
Feat: Add mergeFrom

### PR DESCRIPTION
**Pull request by neebdev**
https://github.com/hwchase17/langchainjs/issues/2136 asked for merge_from feature like in the python version.

### CODE REVIEW

**Comment by neebdev** [comment]
There are some issues. Testing.

**Comment by neebdev** [comment]
Added test. @ewfian Please review and let me know. Thanks

File: src/faiss.cc [hunk 1]
```diff
@@ -63,6 +63,7 @@ class IndexFlatL2 : public Napi::ObjectWrap<IndexFlatL2>
       InstanceMethod("add", &IndexFlatL2::add),
       InstanceMethod("search", &IndexFlatL2::search),
       InstanceMethod("write", &IndexFlatL2::write),
+      InstanceMethod("mergeFrom", &IndexFlatL2::mergeFrom),
```
Comment (ewfian) [inline src/faiss.cc:66]: The types file (lib\index.d.ts)  also need update.

File: src/faiss.cc [hunk 2]
```diff
@@ -254,6 +255,43 @@ class IndexFlatL2 : public Napi::ObjectWrap<IndexFlatL2>
 
     faiss::write_index(index_.get(), fname.c_str());
 
+    return env.Undefined();
+  }
+  Napi::Value mergeFrom(const Napi::CallbackInfo &info)
```
Comment (ewfian) [inline src/faiss.cc:260]: format: add new line before

File: src/faiss.cc [hunk 3]
```diff
@@ -254,6 +255,43 @@ class IndexFlatL2 : public Napi::ObjectWrap<IndexFlatL2>
...
+      Napi::TypeError::New(env, "Invalid argument type, must be an object.").ThrowAsJavaScriptException();
+      return env.Undefined();
+    }
+
+    Napi::Object otherIndex = info[0].As<Napi::Object>();
+    IndexFlatL2 *otherIndexInstance = Napi::ObjectWrap<IndexFlatL2>::Unwrap(otherIndex);
+
+    if (!otherIndexInstance)
+    {
+      Napi::Error::New(env, "Invalid argument type, must be an instance of IndexFlatL2.").ThrowAsJavaScriptException();
```
Comment (ewfian) [inline src/faiss.cc:281]: exceptions also needs to included in unit tests

File: test/IndexFlatL2.test.js [hunk 4]
```diff
@@ -133,4 +133,33 @@ describe('IndexFlatL2', () => {
             expect(index.search([1, 1], 4)).toMatchObject({ distances: [0, 1, 1, 4], labels: [3, 0, 1, 2] });
         });
     });
+    describe("#merge", () => {
```
Comment (ewfian) [inline test/IndexFlatL2.test.js:136]: format: add new line before

**Comment by neebdev** [comment]
@ewfian I added the tests for exceptions, an explicit check for dimensions and also added the type definition for it.

**Comment by ewfian** [comment]
LGTM, Thank you.
will merge in v0.3.0 after v0.2.3 released.

**Pull request merged**